network/p2p/router.go

```diff
@@ -56,26 +51,10 @@ func NewRouter(log logging.Logger, sender common.AppSender) *Router {
 	}
 }
 
-func (r *Router) Connected(_ context.Context, nodeID ids.NodeID, _ *version.Application) error {
-	r.lock.Lock()
-	defer r.lock.Unlock()
-
-	r.peers.Add(nodeID)
-	return nil
-}
-
-func (r *Router) Disconnected(_ context.Context, nodeID ids.NodeID) error {
-	r.lock.Lock()
-	defer r.lock.Unlock()
-
-	r.peers.Remove(nodeID)
-	return nil
-}
-
 // RegisterAppProtocol reserves an identifier for an application protocol and
 // returns a Client that can be used to send messages for the corresponding
 // protocol.
-func (r *Router) RegisterAppProtocol(handlerID uint64, handler Handler) (*Client, error) {
+func (r *Router) RegisterAppProtocol(handlerID uint64, handler Handler, nodeSampler NodeSampler) (*Client, error) {
```

Comment:
We want to be careful about breaking SDK changes. Probably not worth changing now but if you foresee more configurations for the protocols we register might be nice to make this a variadic func or take a config.
```suggestion
(r *Router) RegisterAppProtocol(handlerID uint64, handler Handler, opts ...ProtocolOption) (*Client, error) {
```

Reply:
The options idea is interesting. GRPC seems to have a private `apply` interface to apply the configs. I think something like that seems interesting, I'll look into it.

Reply:
Hmm I don't want to add a variadic options field yet, I think it makes this more complex. I think it's something we definitely want to add later though. We probably don't have any users for this package yet so I'm fine making a breaking change here.

My main concern is that if we make this an options field we need to figure out a way to inject a default node sampler that the option would override. The only way to do this is if we add a `Peers` or `Validators` to the `Router`, which I feel like looks strange since I just want the `Router` to have the responsibility of handling message routing logic.

Reply:
just something to think about for now